src/endpoint/s3/s3_bucket_logging.js

```diff
@@ -0,0 +1,52 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const dbg = require('../../util/debug_module')(__filename);
+const http_utils = require('../../util/http_utils');
+
+function is_bucket_logging_enabled(source_bucket) {
+
+    if (!source_bucket || !source_bucket.bucket_info.logging) {
+
+        return false;
+    }
+
+    return true;
+
+}
+
+function endpoint_bucket_op_logs(op_name, req, res, source_bucket) {
+
+    dbg.log2("Sending op logs for op name = ", op_name);
+    // 1 - Get all the information to be logged in a log message.
+    // 2 - Format it and send it to log bucket/syslog.
+
+    const s3_log = get_bucket_log_record(op_name, source_bucket, req, res);
+    dbg.log1("!!!BUCKET Operation Logs!!! ", s3_log);
+
+}
+
+function get_bucket_log_record(op_name, source_bucket, req, res) {
+
+    const client_ip = http_utils.parse_client_ip(req);
+
+    const log = {
```

Comment:
I think some fields are missing?
-time
-requester (if we have such a concept)
-error code
-bytes_send (if available?)
-object size
etc. see https://docs.aws.amazon.com/AmazonS3/latest/userguide/LogFormat.html

Reply:
Same here. We may not log the exact S3 format. So will decide later.